From aa3424923fdd8de29046357ae364b9a0426786e0 Mon Sep 17 00:00:00 2001
From: svenkatachalam <svenkatachalam@maxlinear.com>
Date: Thu, 22 Aug 2024 13:00:02 +0800
Subject: [PATCH] WLANRTSYS-82831 Enable beacon protection by default for 11BE
 mode.

Issue:
Beacon protection is not enabled by default for 11be mode
in hostap-ng.

Fix:
According to IEEE P802.11be/D3.0, Section 4.3.16a, an
Extremely High Throughput (EHT) STA requires beacon
protection support for an EHT AP. Hence, beacon protection
is enabled when IEEE 802.11be is set for a BSS. Jurijs has
upstreamed the same fix to the community, but it has not
yet been accepted.

FIXUP to WLANRTSYS-60843 MLO changes: AP MLD configuration.
---
 src/ap/ap_config.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/ap/ap_config.c b/src/ap/ap_config.c
index bbb87072e..6474560c9 100644
--- a/src/ap/ap_config.c
+++ b/src/ap/ap_config.c
@@ -1650,6 +1650,12 @@ static int hostapd_config_check_bss(struct hostapd_bss_config *bss,
 		wpa_printf(MSG_INFO,
 			   "Disabling IEEE 802.11be as IEEE 802.11ax is disabled for this BSS");
 	}
+
+	if (full_config && conf->ieee80211be && !bss->disable_11be && !bss->beacon_prot) {
+		bss->beacon_prot = 1;
+		wpa_printf(MSG_INFO,
+			   "Enabling beacon protection as IEEE 802.11be is enabled for this BSS");
+	}
 #endif /* CONFIG_IEEE80211BE */
 
 	if (full_config && bss->ignore_broadcast_ssid && conf->mbssid) {
-- 
2.43.0

